packages/components/src/unit-control/index.tsx

```diff
@@ -228,26 +177,20 @@ function UnitControl(
 		<Root className="components-unit-control-wrapper" style={ style }>
 			<ValueInput
 				aria-label={ label }
-				type={ isPressEnterToChange ? 'text' : 'number' }
```

Comment:
As far as I can tell the use of `'text'` is only to allow input of any character in Chrome (other common browsers don't limit input). With this PR that's no longer a concern. Removing this will default to `'number'` and have the possible benefit of easier validation since text inputs don't factor `min`, `max`, or `step` into validity.

Reply:
> As far as I can tell the use of `'text'` is only to allow input of any character in Chrome

This is an interesting change, to the point that I'd almost split it to a separate PR so that we can focus on testing it in isolation - WDYT @mirka ?

Reply:
Yeah, let's test it separately 👍

Reply:
I've reverted that from this PR but while working on the paste handling (in #40168 ) I found that robust/conventional support will require that the type be `text`. I left a review comment on that PR that may help elucidate.